flybywiresim/a32nx
# feat(clk): add simvars for Clock CHR and ET

Fixes #6049

## Summary of Changes
Adds simvars for the CHR and ET displays in the Clock module, using a value of -1 when no content is displayed.
 * A32NX_CHRONO_ELAPSED_TIME
 * A32NX_CHRONO_ET_ELAPSED_TIME

Naming was chosen with consistency in mind based on the existing simvars and events in Chrono.tsx and ElapsedTime.tsx.

## Screenshots (if necessary)
No visual changes as only new simvars were added.

## References
N/A

## Additional context
Changes were made to include simvars along with the existing functionality without changing the existing state implementation. Using the simvars exclusively in lieu of the React State hooks was considered but introduced visual judder between two consecutive values at times. 

These new simvars enable cockpit builders to keep their Clock hardware displays in sync with FBW's representation.


Discord username (if different from GitHub): AntennaRex#9906

## Testing instructions


1. Ensure the existing CHR and ET features on the Clock continue to behave normally.
2. Enable MSFS 2020 DevMode to inspect the two new simvars (A32NX_CHRONO_ELAPSED_TIME and A32NX_CHRONO_ET_ELAPSED_TIME) to see the seconds elapsed for the Clock's CHR and ET displays, respectively. Each simvar should be set to -1 when the corresponding display on the Clock is blank.


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
